File: nth_fibonacci/nth_fibonacci.py
# Complexity O(n) n - input number,
# Space O(1)
def getNthFib(n):
    prev = 0
    curr = 1
    if n > 1:
        while (n > 2):
            prev, curr = curr, curr + prev
            n -= 1
        return curr
    return prev


if __name__ == "__main__":
    print(getNthFib(5))

File: nth_fibonacci/nthFibonacci.go
package main

import "fmt"

func getFibonacciNumber(n int) int {
	var tmp int

	prev := 0
	curr := 1
	if n > 1 {
		for n > 2 {
			tmp = curr
			curr += prev
			prev = tmp
			n -= 1
		}
		return curr
	}
	return prev
}

func main() {
	var num int

	fmt.Print("Please, input number: ")
	fmt.Scanf("%d", &num)
	fmt.Println(getFibonacciNumber(num))
}
